libkbfs/block_retrieval_queue.go

```diff
@@ -455,7 +559,25 @@ func (brq *blockRetrievalQueue) FinalizeRequest(
 		// Since we created this channel with a buffer size of 1, this won't
 		// block.
 		req.doneCh <- err
+		doneChans = append(doneChans, r.prefetchDoneCh)
+		errChans = append(errChans, r.prefetchErrCh)
 	}
+
+	go func() {
+		// Fan-out the prefetch result to the channel of each individual
+		// request. Since the slice was created before the request mutex was
+		// released, this won't race.
+		var prefetchChans []chan<- struct{}
+		select {
+		case <-prefetchDoneCh:
+			prefetchChans = doneChans
+		case <-prefetchErrCh:
+			prefetchChans = errChans
+		}
+		for _, ch := range prefetchChans {
+			ch <- struct{}{}
```

Comment:
`RequestWithPrefetch`, which appends requests to the retrieval object, seems to allow `nil` channels.  Do you need to check for `nil` here?